src/vcpkg/base/git.cpp

```diff
@@ -150,62 +153,72 @@ namespace vcpkg
                                                const Path& index_file)
     {
         static constexpr StringView args[] = {StringLiteral{"write-tree"}};
-        return run_git_cmd_with_index(context, git_exe, locator, index_file, args).maybe_output;
+        return run_cmd_git_with_index(context, make_git_command(git_exe, locator, args), index_file);
+    }
+
+    bool parse_git_ls_tree_output(DiagnosticContext& context,
+                                  std::vector<GitLSTreeEntry>& target,
+                                  StringView ls_tree_output,
+                                  StringView ls_tree_command)
+    {
+        const auto lines = Strings::split(ls_tree_output, '\0');
+        // The first line of the output is always the parent directory itself.
+        for (auto&& line : lines)
+        {
+            // The default output comes in the format:
+            // <mode> SP <type> SP <object> TAB <file>
+            auto split_line = Strings::split(line, '\t');
```

Comment:
_Technically_ this will fail if a filename has an embedded tab?

Reply:
Probably; I think it would be good to replace all of this and use the `-z` format like I used for the new parser I added. But this is just 'extract method' and I didn't want to touch how it works given that it's been shipping in the product for a long time....

Reply:
(Oh, and yes, overall I find building parsers out of string::split to be an extreme smell.)